localstack/services/apigateway/apigateway_listener.py

```diff
@@ -228,18 +202,43 @@ def forward_request(self, method, path, data, headers):
             elif integration['type'] == 'AWS_PROXY':
                 if uri.startswith('arn:aws:apigateway:') and ':lambda:path' in uri:
                     func_arn = uri.split(':lambda:path')[1].split('functions/')[1].split('/invocations')[0]
+                    account_id = uri.split(':lambda:path')[1].split(':function:')[0].split(':')[-1]
                     data_str = json.dumps(data) if isinstance(data, dict) else data
 
                     relative_path, query_string_params = extract_query_string_params(path=relative_path)
 
+                    source_ip = headers['X-Forwarded-For'].split(',')[-2]
```

Comment:
@whummer do you think that I should return and delete the value? cause I used `X-Forwarded-For` header to be able to pass info from the `generic_proxy` to the backend.

Reply:
Better use this:
```
source_ip = headers['X-Forwarded-For'].rpartition(',')[0]
```
(since the original header may already contain a comma)

Reply:
@whummer Not sure if this is correct.

If you check in generic_proxy this how I construct the header:

if the original header is present and event comma, (e.g 1.2.3.4, 5.6.7.8)
then the final header will look like this

`"1.2.3.4, 5.6.7.8, _client_address_, _server_address_"`

`
        x_forwarded_for = forward_headers.get('X-Forwarded-For')
        if x_forwarded_for:
            x_forwarded_for_list = (x_forwarded_for, client_address, server_address)
        else:
            x_forwarded_for_list = (client_address, server_address)

        forward_headers['X-Forwarded-For'] = ', '.join(x_forwarded_for_list)

`

and if the original header is not present then the final header will look like this

`"_client_address_, _server_address_"`
That's why to return the source_ip I get the address at position [-2]. Does it make sense or am I missing something? Thanks a lot =)

Reply:
Sorry, I missed that. Thanks for clarifying.